gohypergiant/hyperdrive
# MLSDK-446: FastApp | `batch` endpoint is not working

## Summary

_A brief summary of the objective._
The `batch` endpoint for `fastapp`, which is used to request multiple predictions, was broken and needed to be fixed.

_Describe your changes._
Upon hitting the `batch` endpoint, the app will now execute a newly created `batch_predict` function. This function can handle multiple inferences.

## Testing

_Describe steps to follow to verify the changes that were made._
1. Create a `hyperpack` using the `hyperpackage` library. For this example, use a binary classification model.
2. Put the `hyperpack` into a directory, as well as the `study.yaml` that was created in No. 1.
3. Run `hyper pack run` in that directory.
4. During the `hyper pack run` operation, look for the Hyperpackage docker container port number, which will be in the message: `Hyperpackage now running via Docker Container a9cdfe965d on port XXXXX`.
5. During the `hyper pack run` operation, look for the Fast App API key, which will be in the message: `*** Fast App API key is: A_BUNCH_OF_CHARS`.
6. Run this curl command in the same directory, using the info from No. 4 and No. 5:
curl -X 'POST' http://127.0.0.1:[HYPERPACKAGE_DOCKER_CONTAINER_PORT]/batch -H 'accept: application/json' -H 'x-api-key: [FAST_API_KEY]' -d '[[22.6734323 , 133.87953978,  71.25881828,  24.74618134], [3, 28, 15000, 103], [1, 15, 9000, 53]]'. 
7. You should see a list of predictions:
![curl_batch](https://user-images.githubusercontent.com/92936941/190214476-d0ada781-923f-4bb5-8c55-1abe33e7766f.png)

_If this ticket is a low-fi ticket specify what if anything is testable. A
"not testable" is perfectly acceptable in this case._
N/A

_Point out any previously tested functionality that may have been changed
and need another look._
N/A

## Notes

_Optional_

_Extra details about the PR that don't fit elsewhere such as dependencies
on another branch, a known issue that needs extra attention, or links._

## Screenshots

_Optional_

## Ticket(s)

- [MLSDK-446](https://gohypergiant.atlassian.net/browse/MLSDK-446)
